jsm.go

```diff
@@ -884,3 +934,30 @@ func (js *js) StreamNames(ctx context.Context) <-chan string {
 
 	return ch
 }
+
+func getJSContextOpts(defs *js, opts ...JSOpt) (*js, context.CancelFunc, error) {
+	var o js
```

Comment:
feels like maybe this `var o js` should probably should have been a `jsOpts` similar to `pubOpts`/`subOpts`, and the interface should be taking those instead, wdyt?

```diff
- configureJSContext(opts *js)
+ configureJSContext(opts *jsOpts)
```

Reply:
Yeah, makes sense to me. Coming right up.

Reply:
Well, here it is:
https://github.com/nats-io/nats.go/pull/674/commits/105e1aaf2862d60150883c3917e74baec8f797ed

Not sure if we want this here or in another PR.

Reply:
I mean more to split the `type js` which has the behaviors from the options `type jsOpts`.

Reply:
jsOpts commit reverted. Will follow up in new PR. 👍🏽